Ins 30373 (#339)
IBM/universal-connectors: docs/Guardium_Insights/SaaS_1.0/UC_Configuration_GI.md

    
2.  Click **Add connection.**  The Connect to new data source page opens, with a card for each available data source type.
    
3.  Select the data source type. This opens a panel that aids you in initiating the connection.
    
    a **Select data source environment**: Select the environment that hosts your data source.
    
    b. **Select connection method**: Select **Universal Connector.**
    
    c. The remainder of the panel provides **Additional information** about the connection type that you are creating.
    
    d. After reading the **Additional information**, click **Configure**.
    
4.  Enter the details for this connection.
    
    a. In the Name and description page, enter this information:
    
      i. **Name**: A unique name for the connection. This name distinguishes this connection from all other Guardium Insights connections.
    
      ii. **Description**: Enter a description for the connection.
    
    b. Click **Next**.
    
    c. In the Build pipeline page, use the **Choose input plug-in** menu to select your input plug-in. Then select a filter plug-in by using the **Choose a filter plug-in** menu.
    
    d. Click **Next**.
    

>     e. Follow the instructions according to the input plug-in type you selected in <em>**step c**</em>.

Reviewer: ```suggestion
    e. Follow the instructions according to the input plug-in type you selected in **step c**.
```

## JDBC input plug-in configuration

 In the **Additional info** page, specify the following details for the connection you want to create:

*   **Connection string** - the JDBC connection string, for example:
   ```
   jdbc:sqlserver://<server_name>:<port_number>;databaseName=<database_name>
   ```
**NOTE** Do not enter the database user name and password in the connection string, as they are supplied in the **JDBC user** and **Password** fields mentioned in previous steps.

> *   **Statement** - the statement setting determines which audit tables the SELECT query calls for the audit logs. In the Guardium UI, the **Statement*** is divided into three parts to enhance clarity and ease of use: `SELECT` for choosing columns, `FROM` for specifying tables, and `WHERE` for adding filter conditions.

Reviewer: ```suggestion
*   **Statement** - the statement setting determines which audit tables the `SELECT` query calls for the audit logs. In the Guardium UI, the **Statement*** is divided into three parts to enhance clarity and ease of use: `SELECT` for choosing columns, `FROM` for specifying tables, and `WHERE` for adding filter conditions.
```

* **Enrollment ID** - used for an AzureSQL connection. (For more information, click [here](https://github.com/IBM/universal-connectors/tree/main/filter-plugin/logstash-filter-azure-sql-guardium#5-finding-enrollment-id)).
* **Server name** - enter the host name of the database server.

 ## Filebeat input plug-in configuration
### Prerequisite: 

On the data source server, create a **certificate authority** and **certificate** for Filebeat.
These certificates will be used later to establish secured connections between the data source server and universal connector.
1. Download the [create_certificates.sh](create_certificates.sh) script to the datasource server.
2. Change the file permissions so the script can run:
   ```
   chmod +x create_certificates.sh
   ```
3. Run the script with 2 arguments:
  The first argument should be the path where the certificates will be stored and the host name.
   ```
   ./create_certificates.sh <PATH TO STORE> <DATASOURCE SERVER DNS>
   ```
   for example:
   ```
   ./create_certificates.sh /path/to/store datasource.server.dns.com
   ```

4. Copy `filebeatCA.crt` to your local system.

### Configuring a Filebeat connection:

> 1. Exectue steps 1-4 from the [main procedure](../SaaS_1.0/UC_Configuration_GI.md#procedure) . For the data source, select **MongoDB**. For the data source enviornment type, select **On-premises**. For the input plug-in, choose **Filebeat**.

Reviewer: ```suggestion
1. Execute steps 1-4 from the [main procedure](../SaaS_1.0/UC_Configuration_GI.md#procedure) . For the data source, select **MongoDB**. For the data source environment type, select **On-premises**. For the input plug-in, choose **Filebeat**.
```

    
*   **Account ID** (Mandatory) - your AWS account ID (For more information, click [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/console_account-alias.html#FindingYourAWSId)).
    
*   **CloudWatch Log Group name** - specify the log group that is created for your data instance.
    
    For example "/aws/rds/instance/any\_instance/any\_log\_group".
    
    ***NOTE: Due to the possibility of CloudWatch reporting events multiple times for plug-ins configured for Guardium Insights, it is recommended to use SQS instead***

## Azure Event Hubs input plug-in configuration

In the **Additional info** page, specify the details of the connection you want to create:

* **Event hub connections** - specify the list of connection strings that identifies the Event Hubs to be read. Connection strings include the `EntityPath` for the Event Hub.

* **Enrollment ID** - Azure enrollment ID: Unique subscription identifier for billing and resource management.
 ## Configuring Filebeat to forward audit logs to Guardium
    
    
  ## Procedure

> prerequisites:

Reviewer: ```suggestion
### Prerequisites:
```

* On the data source server there is a **certificate authority** (filebeatCA key and certificate).
* A connection was added by the steps in the section: [Filebeat input plug-in configuration](UC_Configuration_GI.md#filebeat-input-plug-in-configuration). 
The `data source tag`, `host` and `UC certificate authority` from the configuration notes are used in these steps.

1.  Open the Filebeat configuration (filebeat.yml) usually located in `/etc/filebeat/filebeat.yml`.
    
2.  Locate the `tags` section and enter the data source tag. For example, `tags: ["any-mongodb"]`.
    
3.  Locate the `output.logstash` section and add an entry for Guardium Insights:
    ```
    # The Logstash hosts
    hosts: ["<hostname-URL>:443"]
    ```

> **NOTE**: In Guardium Insights, whenever using plug-ins that are based on Filebeat as a data shipper, the configured port should be 443. Guardium Insights will map this to an internal port

Reviewer: ```suggestion
**NOTE**: In Guardium Insights, whenever you are using plug-ins that are based on Filebeat as a data shipper, the configured port should be 443. Guardium Insights will map this to an internal port
```
